ui/src/components/directory/topbar/eclSearch/builder/Refinement.vue

```diff
@@ -313,14 +323,31 @@ async function updateValueTreeRoots(): Promise<void> {
 async function updateIsValidProperty(): Promise<void> {
   if (props.focus?.iri === SNOMED.ANY) isValidProperty.value = true;
   else if (props.focus && hasProperty.value) {
-    const request: FunctionRequest = {
-      functionIri: IM_FUNCTION.ALLOWABLE_PROPERTIES,
-      arguments: [
-        { parameter: "focus", valueObject: props.focus },
-        { parameter: "searchIri", valueData: props.value.property.concept?.iri }
-      ]
-    };
-    isValidProperty.value = await FunctionService.runAskFunction(request);
+    const queryRequest: QueryRequest = {
+      query: { "@id": QUERY.ALLOWABLE_PROPERTIES },
+      askIri: selectedProperty.value?.iri
+    } as QueryRequest;
+    if (props.focus.type === "BoolGroup" && isArrayHasLength(props.focus.items)) {
+      const iris: TTIriRef[] = [];
+      for (const item of props.focus.items) {
+        if (item.type === "Concept") iris.push({ "@id": item.concept.iri });
+      }
+      queryRequest.argument = [
+        {
+          parameter: "this",
+          valueIriList: iris
+        }
+      ];
+    } else {
+      queryRequest.argument = [
+        {
+          parameter: "this",
+          valueIri: { "@id": props.focus.iri }
+        }
+      ];
+      osQueryForPropertySearch.value = undefined;
```

Comment:
osQueryForPropertySearch should be set to undefined only from updateQueryForPropertySearch()